common/persistence/cassandra/queue_v2_store.go

```diff
@@ -533,17 +448,28 @@ func GetQueue(
 }
 
 func (s *queueV2Store) getNextMessageID(ctx context.Context, queueType persistence.QueueV2Type, queueName string) (int64, error) {
+	maxMessageID, ok, err := s.getMaxMessageID(ctx, queueType, queueName)
+	if err != nil {
+		return 0, err
+	}
+	if !ok {
+		return persistence.FirstQueueMessageID, nil
+	}
+
+	// The next message ID is the max message ID + 1.
+	return maxMessageID + 1, nil
+}
+
+func (s *queueV2Store) getMaxMessageID(ctx context.Context, queueType persistence.QueueV2Type, queueName string) (int64, bool, error) {
 	var maxMessageID int64
 
 	err := s.session.Query(TemplateGetMaxMessageIDQuery, queueType, queueName, 0).WithContext(ctx).Scan(&maxMessageID)
 	if err != nil {
 		if gocql.IsNotFoundError(err) {
 			// There are no messages in the queue, so the next message ID is the first message ID.
```

Comment:
This comment no longer applies. We're returning false here, so the first return value is ignored.

Reply:
Done